### PR TITLE
Feat/email validation pages in frontend

### PR DESCRIPTION
**Pull request by SilviaPabon**
You can try this pr, along with the (https://github.com/PedroChaparro/loomies-backend/pull/62) feat/email-validation backend branch, or if it's already approved, you can try with the dev branch.

There are two new views:
- Validate account: It is used to validate the code that is sent by email. It has a space for the email and for the code.
image

- Send new code: It is used to send a new mail to the mail. It has a space for mail.
image

In the different views, links are offered to go to different points as needed, for example, if someone is already sign up and is in login page, but their code is not validated, they can request to do so by clicking on validate account.

### CODE REVIEW

File: src/pages/NewCodeView.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      } else {
+        showSuccessToast(response?.message);
+        redirectToEmailVal();
+      }
+    }
+  });
+  return (
+    <View style={Styles.container}>
+      <View style={Styles.header}>
+        <Text style={Styles.headerTitle}>INSERT YOUR EMAIL</Text>
```
Comment (PedroChaparro) [inline src/pages/NewCodeView.tsx:39]: The main title of the view should be something like "Request a new code", so the user knows what the purpose of the view is.

Comment (SilviaPabon) [reply src/pages/NewCodeView.tsx]: You are right, I fix it!

**Review by Woynert — COMMENTED** [review]
May be a problem with my setup. But every time I press the **validate account** button in the login screen the app stops with an error. Can someone try it?

```
 ERROR  TypeError: undefined is not an object (evaluating 'route.params.email')

This error is located at:
    in EmailValidationView (at SceneView.tsx:132)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:124)
    in SceneView (at useDescriptors.tsx:217)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:281)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:279)
    in RCTView (at View.js:32)
    in View (at CardSheet.tsx:45)
    in CardSheet (at Card.tsx:557)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at Card.tsx:536)
    in PanGestureHandler (at GestureHandlerNative.tsx:14)
    in PanGestureHandler (at Card.tsx:530)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at Card.tsx:526)
    in RCTView (at View.js:32)
    in View (at Card.tsx:520)
    in Card (at CardContainer.tsx:219)
    in CardContainer (at CardStack.tsx:669)
    in RNSScreen (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at src/index.native.tsx:266)
    in Suspender (at src/index.tsx:40)
    in Suspense (at src/index.tsx:39)
    in Freeze (at src/index.native.tsx:169)
    in DelayedFreeze (at src/index.native.tsx:265)
    in InnerScreen (at src/index.native.tsx:445)
    in Screen (at Screens.tsx:38)
    in MaybeScreen (at CardStack.tsx:661)
    in RNSScreenContainer (at src/index.native.tsx:343)
    in ScreenContainer (at Screens.tsx:20)
    in MaybeScreenContainer (at CardStack.tsx:572)
    in RCTView (at View.js:32)
    in View (at Background.tsx:13)
    in Background (at CardStack.tsx:570)
    in CardStack (at StackView.tsx:437)
    in RNCSafeAreaProvider (at SafeAreaContext.tsx:87)
    in SafeAreaProvider (at SafeAreaProviderCompat.tsx:46)
    in SafeAreaProviderCompat (at StackView.tsx:430)
    in RNGestureHandlerRootView (at GestureHandlerRootView.android.tsx:23)
    in GestureHandlerRootView (at StackView.tsx:429)
    in StackView (at createStackNavigator.tsx:121)
    in PreventRemoveProvider (at useNavigationBuilder.tsx:713)
    in NavigationContent (at useComponent.tsx:35)
    in Unknown (at createStackNavigator.tsx:120)
    in StackNavigator (at StackNavigation.tsx:18)
    in MapProvider (at StackNavigation.tsx:17)
    in StackNavigation (at App.tsx:11)
    in EnsureSingleNavigator (at BaseNavigationContainer.tsx:430)
    in BaseNavigationContainer (at NavigationContainer.tsx:132)
    in ThemeProvider (at NavigationContainer.tsx:131)
    in NavigationContainerInner (at App.tsx:10)
    in AuthProvider (at App.tsx:9)
    in App (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in Loomies(RootComponent) (at renderApplication.js:60)
 ERROR  TypeError: undefined is not an object (evaluating 'route.params.email')

This error is located at:
    in EmailValidationView (at SceneView.tsx:132)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:124)
    in SceneView (at useDescriptors.tsx:217)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:281)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:279)
    in RCTView (at View.js:32)
    in View (at CardSheet.tsx:45)
    in CardSheet (at Card.tsx:557)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at Card.tsx:536)
    in PanGestureHandler (at GestureHandlerNative.tsx:14)
    in PanGestureHandler (at Card.tsx:530)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at Card.tsx:526)
    in RCTView (at View.js:32)
    in View (at Card.tsx:520)
    in Card (at CardContainer.tsx:219)
    in CardContainer (at CardStack.tsx:669)
    in RNSScreen (at createAnimatedComponent.js:211)
    in AnimatedComponent (at createAnimatedComponent.js:264)
    in AnimatedComponentWrapper (at src/index.native.tsx:266)
    in Suspender (at src/index.tsx:40)
    in Suspense (at src/index.tsx:39)
    in Freeze (at src/index.native.tsx:169)
    in DelayedFreeze (at src/index.native.tsx:265)
    in InnerScreen (at src/index.native.tsx:445)
    in Screen (at Screens.tsx:38)
    in MaybeScreen (at CardStack.tsx:661)
    in RNSScreenContainer (at src/index.native.tsx:343)
    in ScreenContainer (at Screens.tsx:20)
    in MaybeScreenContainer (at CardStack.tsx:572)
    in RCTView (at View.js:32)
    in View (at Background.tsx:13)
    in Background (at CardStack.tsx:570)
    in CardStack (at StackView.tsx:437)
    in RNCSafeAreaProvider (at SafeAreaContext.tsx:87)
    in SafeAreaProvider (at SafeAreaProviderCompat.tsx:46)
    in SafeAreaProviderCompat (at StackView.tsx:430)
    in RNGestureHandlerRootView (at GestureHandlerRootView.android.tsx:23)
    in GestureHandlerRootView (at StackView.tsx:429)
    in StackView (at createStackNavigator.tsx:121)
    in PreventRemoveProvider (at useNavigationBuilder.tsx:713)
    in NavigationContent (at useComponent.tsx:35)
    in Unknown (at createStackNavigator.tsx:120)
    in StackNavigator (at StackNavigation.tsx:18)
    in MapProvider (at StackNavigation.tsx:17)
    in StackNavigation (at App.tsx:11)
    in EnsureSingleNavigator (at BaseNavigationContainer.tsx:430)
    in BaseNavigationContainer (at NavigationContainer.tsx:132)
    in ThemeProvider (at NavigationContainer.tsx:131)
    in NavigationContainerInner (at App.tsx:10)
    in AuthProvider (at App.tsx:9)
    in App (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in Loomies(RootComponent) (at renderApplication.js:60)
undefined
```

**Comment by PedroChaparro** [comment]
> May be a problem with my setup. But every time I press the **validate account** button in the login screen the app stops with an error. Can someone try it?
> 

It seems like the login page doesn't send the email param to the validation view @SilviaPabon, my bad.

**Comment by PedroChaparro** [comment]
> May be a problem with my setup. But every time I press the **validate account** button in the login screen the app stops with an error. Can someone try it?
> 

It should work now @Woynert.

**Pull request merged**